src/runtime/CMakeLists.txt

```diff
@@ -21,85 +23,103 @@ set(HIPSYCL_SOURCES
+  else()
+    # Supress warnings because wrongly set CXX arguments
+    target_compile_options(rt-backend-hip PRIVATE -Wno-unused-command-line-argument)
+    target_link_libraries(rt-backend-hip PRIVATE hipSYCL-rt  hip::host)
+  endif()
+
+  install(TARGETS rt-backend-hip
+        EXPORT install_exports
+        LIBRARY DESTINATION lib/hipSYCL
+        ARCHIVE DESTINATION lib/static)
 endif()
 
 if(WITH_CPU_BACKEND)
-  set(HIPSYCL_SOURCES ${HIPSYCL_SOURCES}
+  add_library(rt-backend-omp SHARED
     omp/omp_allocator.cpp
     omp/omp_backend.cpp
     omp/omp_event.cpp
     omp/omp_hardware_manager.cpp
     omp/omp_queue.cpp)
-endif()
 
-add_library(hipSYCL-rt SHARED ${HIPSYCL_SOURCES})
+    find_package(OpenMP REQUIRED)
 
-# syclcc already knows about these include directories, but clangd-based tooling does not.
-# Specifying them explicitly ensures that IDEs can resolve all hipSYCL includes correctly.
-target_include_directories(hipSYCL-rt
-  PUBLIC
-    $<BUILD_INTERFACE:${HIPSYCL_SOURCE_DIR}/include>
-    $<INSTALL_INTERFACE:include>
-  PRIVATE
-    ${HIPSYCL_SOURCE_DIR}
-)
+    target_include_directories(rt-backend-omp PRIVATE ${HIPSYCL_SOURCE_DIR}/include)
+    if(APPLE)
+      if(CMAKE_VERSION VERSION_LESS "3.16")
+        message(FATAL_ERROR "CMake 3.16.0+ is required for macOS OpenMP support!")
+      endif()
+      target_include_directories(rt-backend-omp PRIVATE ${OpenMP_CXX_INCLUDE_DIRS})
+    endif()
 
-if(WITH_CUDA_BACKEND)
-  target_compile_definitions(hipSYCL-rt PRIVATE HIPSYCL_RT_ENABLE_CUDA_BACKEND=1)
-  target_include_directories(hipSYCL-rt PRIVATE ${CUDA_TOOLKIT_ROOT_DIR}/include)
-  
-  target_link_libraries(hipSYCL-rt PRIVATE ${CUDA_LIBRARIES} ${CUDA_DRIVER_LIBRARY})
-endif()
+    target_link_libraries(rt-backend-omp PRIVATE hipSYCL-rt  OpenMP::OpenMP_CXX)
 
-if(WITH_ROCM_BACKEND) 
-  target_compile_definitions(hipSYCL-rt PRIVATE HIPSYCL_RT_ENABLE_HIP_BACKEND=1 HIPSYCL_RT_HIP_TARGET_ROCM=1)
-  if(NOT HIP_FOUND)
-    target_include_directories(hipSYCL-rt PRIVATE ${ROCM_PATH}/include)
-    target_link_libraries(hipSYCL-rt PRIVATE ${ROCM_LIBS})
-  else()
-    # Supress warnings because wrongly set CXX arguments
-    target_compile_options(hipSYCL-rt PRIVATE -Wno-unused-command-line-argument)
-    target_link_libraries(hipSYCL-rt PRIVATE hip::host)
-  endif()
+    install(TARGETS rt-backend-omp
+        EXPORT install_exports
+        LIBRARY DESTINATION lib/hipSYCL
+        ARCHIVE DESTINATION lib/static)
 endif()
 
-if(WITH_CPU_BACKEND)
-  find_package(OpenMP REQUIRED)
-  if(APPLE)
-    if(CMAKE_VERSION VERSION_LESS "3.16")
-      message(FATAL_ERROR "CMake 3.16.0+ is required for macOS OpenMP support!")
-    endif()
-    target_include_directories(hipSYCL-rt PRIVATE ${OpenMP_CXX_INCLUDE_DIRS})
-  endif()
-
-  target_link_libraries(hipSYCL-rt PRIVATE OpenMP::OpenMP_CXX)
 
-  target_compile_definitions(hipSYCL-rt PRIVATE HIPSYCL_RT_ENABLE_OMP_BACKEND=1)
+if(WITH_CPU_BACKEND)
```

Comment:
forgot something here?

Reply:
whopsie - forgot to delete that old `if()` ;) Removed.